File: test_upload.dart
import 'dart:convert';
import 'dart:io';
import 'package:http/http.dart' as http;

Future<void> main() async {
  print('Testing Catbox...');
  
  // Create a dummy image
  final dummyBytes = List<int>.generate(100, (i) => i);
  
  try {
    final request = http.MultipartRequest("POST", Uri.parse("https://catbox.moe/user/api.php"));
    request.fields['reqtype'] = 'fileupload';
    request.fields['userhash'] = '';
    
    // Add user agent
    // request.headers['User-Agent'] = 'Mozilla/5.0';
    
    final multipartFile = http.MultipartFile.fromBytes(
      'fileToUpload',
      dummyBytes,
      filename: "test.jpg",
    );
    request.files.add(multipartFile);

    final response = await request.send();
    final responseBody = await response.stream.bytesToString();

    print('Status: \${response.statusCode}');
    print('Body: \$responseBody');
  } catch (e) {
    print('Error: \$e');
  }
}

File: test_upload.py
import urllib.request
import urllib.parse
from urllib.error import URLError

boundary = '----WebKitFormBoundary7MA4YWxkTrZu0gW'
body = (
    '--' + boundary + '\r\n'
    'Content-Disposition: form-data; name="reqtype"\r\n\r\n'
    'fileupload\r\n'
    '--' + boundary + '\r\n'
    'Content-Disposition: form-data; name="fileToUpload"; filename="test.jpg"\r\n'
    'Content-Type: image/jpeg\r\n\r\n'
    'testdata\r\n'
    '--' + boundary + '--\r\n'
)

req = urllib.request.Request('https://catbox.moe/user/api.php', data=body.encode('utf-8'))
req.add_header('Content-Type', f'multipart/form-data; boundary={boundary}')
req.add_header('User-Agent', 'Mozilla/5.0')
try:
    with urllib.request.urlopen(req) as response:
        print(response.read().decode())
except Exception as e:
    print(e)
